bitcoin/core/key.py

```diff
@@ -28,6 +28,13 @@
 # this specifies the curve used with ECDSA.
 _NID_secp256k1 = 714 # from openssl/obj_mac.h
 
+# test that openssl support secp256k1
+if _ssl.EC_KEY_new_by_curve_name(_NID_secp256k1).value is None:
+    errno = _ssl.ERR_get_error()
+    errmsg = ctypes.create_string_buffer(120)
+    _ssl.ERR_error_string_n(errno, errmsg, 120)
+    raise Exception('openssl error: %s' % errmsg.value)
```

Comment:
Might make sense to change this to a RuntimeError(), from the Py2/3 docs:

"Raised when an error is detected that doesn’t fall in any of the other categories. The associated value is a string indicating what precisely went wrong."

When this error is triggered, what does it look like? IE, what does errmsg.value get set too? (I have no easy way to test this)